src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -467,9 +469,15 @@ impl Coordinator {
         let ingestion_id = source.global_id();
         let subsource_stmts = generate_subsource_statements(&scx, source_name, subsources)?;
 
+        let id_ts = self.get_catalog_write_ts().await;
+        let mut ids = self
+            .catalog_mut()
+            .allocate_user_ids(u64::cast_from(subsource_stmts.len()), id_ts)
+            .await?;
         for subsource_stmt in subsource_stmts {
+            let (item_id, global_id) = ids.pop().unwrap();
             let s = self
-                .plan_subsource(session, &params, subsource_stmt)
+                .plan_subsource(session, &params, subsource_stmt, item_id, global_id)
```

Comment:
This changes the order of the IDs .. not sure if it's relevant!

Reply:
We've been bitten by ID ordering before, would be great to retain the existing order. See my other comment related to using `.zip(...)`